app/org/sagebionetworks/bridge/play/controllers/AuthenticationController.java

```diff
@@ -32,26 +35,27 @@
 @Controller
 public class AuthenticationController extends BaseController {
 
-    public Result requestEmailSignIn() { 
+    public Result requestEmailSignIn() throws Exception { 
         SignIn signInRequest = parseJson(request(), SignIn.class);
         
         authenticationService.requestEmailSignIn(signInRequest);
         
         return acceptedResult("Email sent.");
     }
     
-    public Result emailSignIn() { 
-        SignIn signInRequest = parseJson(request(), SignIn.class);
+    public Result emailSignIn() throws Exception { 
+        SignIn tokenHolder = parseJson(request(), SignIn.class);
 
-        if (isBlank(signInRequest.getStudyId())) {
-            throw new BadRequestException("Study identifier is required.");
-        }
-        Study study = studyService.getStudy(signInRequest.getStudyId());
-        verifySupportedVersionOrThrowException(study);
+        // We'll provide the study once we reconstitute the transaction
+        ClientInfo clientInfo = getClientInfoFromUserAgentHeader();
+        LinkedHashSet<String> langs = getLanguagesFromAcceptLanguageHeader();
         
-        CriteriaContext context = getCriteriaContext(study.getStudyIdentifier());
+        UserSession session = authenticationService.emailSignIn(clientInfo, langs, tokenHolder);
```

Comment:
Need to capture consent exception, and throw the kill switch exception instead, if necessary, or else re throw the consent exception. Just to be consistent with other methods